### PR TITLE
[FLINK-18824][format][canal] Support serialization for canal-json format

### PR DESCRIPTION
**Pull request by caozhen1937**
## What is the purpose of the change

*Support serialization for canal-json format*

## Brief change log

  - add SerializationSchema for serialize data to byte[]
  - add formatfactory for SerializationSchema

## Verifying this change

  - Added java tests for  serialize data to byte[] into table sink

### CODE REVIEW

**Comment by caozhen1937** [comment]
@flinkbot run azure

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/canal/CanalJsonFormatFactory.java [hunk 1]
```diff
@@ -56,16 +57,16 @@
 	@SuppressWarnings("unchecked")
 	@Override
 	public DecodingFormat<DeserializationSchema<RowData>> createDecodingFormat(
-			DynamicTableFactory.Context context,
-			ReadableConfig formatOptions) {
+		DynamicTableFactory.Context context,
+		ReadableConfig formatOptions) {
```
Comment (wuchong) [inline flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/canal/CanalJsonFormatFactory.java:61]: Please keep indent for the method parameters. 
Same to the following mehods.

File: flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/canal/CanalJsonSerDeserSchemaTest.java [hunk 2]
```diff
@@ -131,27 +131,72 @@ public void testDeserialization() throws Exception {
 			"-D(102,car battery,12V car battery,5.17)",
 			"-D(103,12-pack drill bits,12-pack of drill bits with sizes ranging from #40 to #3,0.8)"
 		);
-		assertEquals(expected, collector.list);
+		List<String> actual = collector.list.stream()
+			.map(Object::toString)
+			.collect(Collectors.toList());
+		assertEquals(expected, actual);
+
```
Comment (wuchong) [inline flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/canal/CanalJsonSerDeserSchemaTest.java:138]: remove empty line.

**Review by wuchong — COMMENTED** [review]
Thanks for the contribution @caozhen1937 , the changes looks good to me in general. I left some minor comments below. 

Besides, could you also update the documentation to introduce this new feature?

**Comment by caozhen1937** [comment]
ok, I will update the documentation to introduce this new feature @wuchong

**Comment by caozhen1937** [comment]
PTAL  @wuchong

**Review by wuchong — APPROVED** [review]
LGTM. I modifed some parts of the code.

Will merge it once build is passed.

**Comment by wuchong** [comment]
Merging...

**Pull request merged**